### ISSUE TITLE
WSL Hangs on launch (dupe / re-open of #6980 with logs)

### ISSUE THREAD

**Issue by jodavaho** [microsoft/WSL#7096]
### Windows Build Number

Microsoft Windows [Version 10.0.21390.2025]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

Kernel version: 5.10.16

### Distro Version

Ubuntu 20.04

### Repro Steps

Launch wsl application, via the WSL for Linux 

### Expected Behavior

WSL launches

### Actual Behavior

WSL freezes

### Diagnostic Logs

https://aka.ms/AAczgdt starting up WSL ubuntu

**Comment by jodavaho** [comment]
From #6980 , logs are at https://aka.ms/AAczgdt

**Comment by benhillis** [comment]
@jodavaho - I do not have access to that link. Can you attach logs directly?

**Comment by OneBlue** [comment]
I have access to those logs.

@jodavaho can you please enable the debug console and share its output ? 

Add this to .wslconfig:

```
[wsl2]
debugConsole=true
```

Then run wsl --shutdown and restart wsl.

**Comment by jodavaho** [comment]
Happy to.  Where do I put that file?

**Comment by jodavaho** [comment]
@OneBlue Requesting completed instructions.

**Comment by therealkenc** [comment]
In `%HOMEPATH%` aka `C:\Users\you`. Docs [here](https://docs.microsoft.com/en-us/windows/wsl/wsl-config#configure-global-options-with-wslconfig).

**Comment by jodavaho** [comment]
And then go through the capture steps when it freezes? Will do.

**Comment by jodavaho** [comment]
OK, I captured a successful startup. I'll post a failed on next time it hangs.

**Comment by jodavaho** [comment]
Here's a frozen / hanging startup log:

```
[    0.000000] Linux version 5.10.16.3-microsoft-standard-WSL2 (oe-user@oe-host) (x86_64-msft-linux-gcc (GCC) 9.3.0, GNU ld (GNU Binutils) 2.34.0.20200220) #1 SMP Fri Apr 2 22:23:49 UTC 2021
[    0.000000] Command line: initrd=\initrd.img panic=-1 nr_cpus=16 swiotlb=force console=ttyS0,115200 debug pty.legacy_count=0
[    0.000000] KERNEL supported cpus:
[    0.000000]   Intel GenuineIntel
[    0.000000]   AMD AuthenticAMD
[    0.000000]   Centaur CentaurHauls
[    0.000000] x86/fpu: Supporting XSAVE feature 0x001: 'x87 floating point registers'
[    0.000000] x86/fpu: Supporting XSAVE feature 0x002: 'SSE registers'
[    0.000000] x86/fpu: Supporting XSAVE feature 0x004: 'AVX registers'
[    0.000000] x86/fpu: xstate_offset[2]:  576, xstate_sizes[2]:  256
[    0.000000] x86/fpu: Enabled xstate features 0x7, context size is 832 bytes, using 'compacted' format.
[    0.000000] BIOS-provided physical RAM map:
[    0.000000] BIOS-e820: [mem 0x0000000000000000-0x000000000009ffff] usable
[    0.000000] BIOS-e820: [mem 0x00000000000e0000-0x00000000000e0fff] reserved
[    0.000000] BIOS-e820: [mem 0x0000000000100000-0x00000000001fffff] ACPI data
[    0.000000] BIOS-e820: [mem 0x0000000000200000-0x00000000f7ffffff] usable
[    0.000000] BIOS-e820: [mem 0x0000000100000000-0x0000000803dfffff] usable
[    0.000000] NX (Execute Disable) protection: active
[    0.000000] DMI not present or invalid.
[    0.000000] Hypervisor detected: Microsoft Hyper-V
[    0.000000] Hyper-V: features 0x2e7f, privilege high: 0x3b8030, hints 0x24c2c, misc 0xe0bed7b2
[    0.000000] Hyper-V Host Build:21390-10.0-2-0.2025
[    0.000000] Hyper-V: LAPIC Timer Frequency: 0x1e8480
[    0.000000] tsc: Marking TSC unstable due to running on Hyper-V
[    0.000000] Hyper-V: Using hypercall for remote TLB flush
[    0.000000] clocksource: hyperv_clocksource_tsc_page: mask: 0xffffffffffffffff max_cycles: 0x24e6a1710, max_idle_ns: 440795202120 ns
[    0.000002] tsc: Detected 2400.010 MHz processor
[    0.000011] e820: update [mem 0x00000000-0x00000fff] usable ==> reserved
[    0.000014] e820: remove [mem 0x000a0000-0x000fffff] usable
[    0.000016] last_pfn = 0x803e00 max_arch_pfn = 0x400000000
[    0.000034] MTRR default type: uncachable
[    0.000035] MTRR fixed ranges enabled:
[    0.000036]   00000-3FFFF write-back
[    0.000036]   40000-7FFFF uncachable
[    0.000037]   80000-8FFFF write-back
[    0.000038]   90000-FFFFF uncachable
[    0.000039] MTRR variable ranges enabled:
[    0.000040]   0 base 0000000000 mask 7F00000000 write-back
[    0.000041]   1 base 0100000000 mask 7000000000 write-back
[    0.000041]   2 disabled
[    0.000042]   3 disabled
[    0.000042]   4 disabled
[    0.000043]   5 disabled
[    0.000043]   6 disabled
[    0.000044]   7 disabled
[    0.000053] x86/PAT: Configuration [0-7]: WB  WC  UC- UC  WB  WP  UC- WT
[    0.000065] last_pfn = 0xf8000 max_arch_pfn = 0x400000000
[    0.000077] Using GB pages for direct mapping
[    0.000552] RAMDISK: [mem 0x03035000-0x03043fff]
[    0.000558] ACPI: Early table checksum verification disabled
[    0.000570] ACPI: RSDP 0x00000000000E0000 000024 (v02 VRTUAL)
[    0.000574] ACPI: XSDT 0x0000000000100000 000044 (v01 VRTUAL MICROSFT 00000001 MSFT 00000001)
[    0.000580] ACPI: FACP 0x0000000000101000 000114 (v06 VRTUAL MICROSFT 00000001 MSFT 00000001)
[    0.000586] ACPI: DSDT 0x00000000001011B8 01E184 (v02 MSFTVM DSDT01   00000001 MSFT 05000000)
[    0.000590] ACPI: FACS 0x0000000000101114 000040
[    0.000593] ACPI: OEM0 0x0000000000101154 000064 (v01 VRTUAL MICROSFT 00000001 MSFT 00000001)
[    0.000596] ACPI: SRAT 0x000000000011F33C 0003B0 (v02 VRTUAL MICROSFT 00000001 MSFT 00000001)
[    0.000600] ACPI: APIC 0x000000000011F6EC 0000C8 (v04 VRTUAL MICROSFT 00000001 MSFT 00000001)
[    0.000607] ACPI: Local APIC address 0xfee00000
[    0.000867] Zone ranges:
[    0.000868]   DMA      [mem 0x0000000000001000-0x0000000000ffffff]
[    0.000869]   DMA32    [mem 0x0000000001000000-0x00000000ffffffff]
[    0.000871]   Normal   [mem 0x0000000100000000-0x0000000803dfffff]
[    0.000884]   Device   empty
[    0.000886] Movable zone start for each node
[    0.000886] Early memory node ranges
[    0.000887]   node   0: [mem 0x0000000000001000-0x000000000009ffff]
[    0.000888]   node   0: [mem 0x0000000000200000-0x00000000f7ffffff]
[    0.000890]   node   0: [mem 0x0000000100000000-0x0000000803dfffff]
[    0.001668] Zeroed struct page in unavailable ranges: 17249 pages
[    0.001670] Initmem setup node 0 [mem 0x0000000000001000-0x0000000803dfffff]
[    0.001674] On node 0 totalpages: 8371359
[    0.001675]   DMA zone: 59 pages used for memmap
[    0.001676]   DMA zone: 22 pages reserved
[    0.001678]   DMA zone: 3743 pages, LIFO batch:0
[    0.001708]   DMA32 zone: 16320 pages used for memmap
[    0.001710]   DMA32 zone: 1011712 pages, LIFO batch:63
[    0.013831]   Normal zone: 114936 pages used for memmap
[    0.013835]   Normal zone: 7355904 pages, LIFO batch:63
[    0.014244] ACPI: Local APIC address 0xfee00000
[    0.014251] ACPI: LAPIC_NMI (acpi_id[0x01] dfl dfl lint[0x1])
[    0.014506] IOAPIC[0]: apic_id 16, version 17, address 0xfec00000, GSI 0-23
[    0.014510] ACPI: INT_SRC_OVR (bus 0 bus_irq 9 global_irq 9 high level)
[    0.014513] ACPI: IRQ9 used by override.
[    0.014515] Using ACPI (MADT) for SMP configuration information
[    0.014522] smpboot: Allowing 16 CPUs, 0 hotplug CPUs
[    0.014530] [mem 0xf8000000-0xffffffff] available for PCI devices
[    0.014532] Booting paravirtualized kernel on Hyper-V
[    0.014534] clocksource: refined-jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.019404] setup_percpu: NR_CPUS:256 nr_cpumask_bits:256 nr_cpu_ids:16 nr_node_ids:1
[    0.020056] percpu: Embedded 52 pages/cpu s173272 r8192 d31528 u262144
[    0.020062] pcpu-alloc: s173272 r8192 d31528 u262144 alloc=1*2097152
[    0.020063] pcpu-alloc: [0] 00 01 02 03 04 05 06 07 [0] 08 09 10 11 12 13 14 15
[    0.020080] Built 1 zonelists, mobility grouping on.  Total pages: 8240022
[    0.020081] Kernel command line: initrd=\initrd.img panic=-1 nr_cpus=16 swiotlb=force console=ttyS0,115200 debug pty.legacy_count=0
[    0.024498] Dentry cache hash table entries: 4194304 (order: 13, 33554432 bytes, linear)
[    0.026667] Inode-cache hash table entries: 2097152 (order: 12, 16777216 bytes, linear)
[    0.026806] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.044979] Memory: 4094124K/33485436K available (16403K kernel code, 2459K rwdata, 3464K rodata, 1444K init, 1164K bss, 675724K reserved, 0K cma-reserved)
[    0.045022] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=16, Nodes=1
[    0.045029] ftrace: allocating 49613 entries in 194 pages
[    0.061388] ftrace: allocated 194 pages with 3 groups
[    0.061806] rcu: Hierarchical RCU implementation.
[    0.061809] rcu:     RCU restricting CPUs from NR_CPUS=256 to nr_cpu_ids=16.
[    0.061810]  Rude variant of Tasks RCU enabled.
[    0.061811]  Tracing variant of Tasks RCU enabled.
[    0.061812] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.061814] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=16
[    0.066415] Using NULL legacy PIC
[    0.066418] NR_IRQS: 16640, nr_irqs: 552, preallocated irqs: 0
[    0.066931] random: crng done (trusting CPU's manufacturer)
[    0.066960] Console: colour dummy device 80x25
[    0.332673] printk: console [ttyS0] enabled
[    0.334293] ACPI: Core revision 20200925
[    0.335960] Failed to register legacy timer interrupt
[    0.338056] APIC: Switch to symmetric I/O mode setup
[    0.340068] Switched APIC routing to physical flat.
[    0.342467] Hyper-V: Using IPI hypercalls
[    0.344801] Hyper-V: Using enlightened APIC (xapic mode)
[    0.344887] Calibrating delay loop (skipped), value calculated using timer frequency.. 4800.02 BogoMIPS (lpj=24000100)
[    0.352013] pid_max: default: 32768 minimum: 301
[    0.354028] LSM: Security Framework initializing
[    0.354884] Mount-cache hash table entries: 65536 (order: 7, 524288 bytes, linear)
[    0.354884] Mountpoint-cache hash table entries: 65536 (order: 7, 524288 bytes, linear)
[    0.354884] Last level iTLB entries: 4KB 64, 2MB 8, 4MB 8
[    0.354884] Last level dTLB entries: 4KB 64, 2MB 0, 4MB 0, 1GB 4
[    0.354884] Spectre V1 : Mitigation: usercopy/swapgs barriers and __user pointer sanitization
[    0.354884] Spectre V2 : Mitigation: Enhanced IBRS
[    0.354884] Spectre V2 : Spectre v2 / SpectreRSB mitigation: Filling RSB on context switch
[    0.354884] Spectre V2 : mitigation: Enabling conditional Indirect Branch Prediction Barrier
[    0.354884] Speculative Store Bypass: Mitigation: Speculative Store Bypass disabled via prctl and seccomp
[    0.354884] TAA: Mitigation: TSX disabled
[    0.354884] SRBDS: Mitigation: TSX disabled
[    0.354884] Freeing SMP alternatives memory: 52K
[    0.354884] smpboot: CPU0: Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz (family: 0x6, model: 0x9e, stepping: 0xd)
[    0.354884] Performance Events: unsupported p6 CPU model 158 no PMU driver, software events only.
[    0.354921] rcu: Hierarchical SRCU implementation.
[    0.357793] smp: Bringing up secondary CPUs ...
[    0.360414] x86: Booting SMP configuration:
[    0.362544] .... node  #0, CPUs:        #1  #2  #3  #4  #5  #6  #7  #8  #9 #10 #11 #12 #13 #14 #15
[    0.365632] smp: Brought up 1 node, 16 CPUs
[    0.376578] smpboot: Max logical packages: 1
[    0.378216] smpboot: Total of 16 processors activated (76800.32 BogoMIPS)
[    0.407480] node 0 deferred pages initialised in 30ms
[    0.407480] devtmpfs: initialized
[    0.407480] x86/mm: Memory block size: 128MB
[    0.415661] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.418606] futex hash table entries: 4096 (order: 6, 262144 bytes, linear)
[    0.421795] NET: Registered protocol family 16
[    0.423592] thermal_sys: Registered thermal governor 'step_wise'
[    0.423616] cpuidle: using governor menu
[    0.423616] ACPI: bus type PCI registered
[    0.423616] PCI: Fatal: No config space access function found
[    0.426010] HugeTLB registered 1.00 GiB page size, pre-allocated 0 pages
[    0.427503] HugeTLB registered 2.00 MiB page size, pre-allocated 0 pages
[    0.434978] raid6: skip pq benchmark and using algorithm avx2x4
[    0.437241] raid6: using avx2x2 recovery algorithm
[    0.439091] ACPI: Added _OSI(Module Device)
[    0.440708] ACPI: Added _OSI(Processor Device)
[    0.442420] ACPI: Added _OSI(3.0 _SCP Extensions)
[    0.444233] ACPI: Added _OSI(Processor Aggregator Device)
[    0.444889] ACPI: Added _OSI(Linux-Dell-Video)
[    0.446610] ACPI: Added _OSI(Linux-Lenovo-NV-HDMI-Audio)
[    0.448647] ACPI: Added _OSI(Linux-HPI-Hybrid-Graphics)
[    0.454577] ACPI: 1 ACPI AML tables successfully acquired and loaded
[    0.465603] ACPI: Interpreter enabled
[    0.467012] ACPI: (supports S0 S5)
[    0.468288] ACPI: Using IOAPIC for interrupt routing
[    0.470171] PCI: Using host bridge windows from ACPI; if necessary, use "pci=nocrs" and report a bug
[    0.473707] ACPI: Enabled 1 GPEs in block 00 to 0F
[    0.475668] iommu: Default domain type: Translated
[    0.477597] SCSI subsystem initialized
[    0.479062] hv_vmbus: Vmbus version:5.2
[    0.479062] PCI: Using ACPI for IRQ routing
[    0.484922] hv_vmbus: Unknown GUID: c376c1c3-d276-48d2-90a9-c04748072c60
[    0.479062] PCI: System does not support PCI
[    0.489088] hv_vmbus: Unknown GUID: 6e382d18-3336-4f4b-acc4-2b7703d4df4a
[    0.489088] hv_vmbus: Unknown GUID: dde9cbc0-5060-4436-9448-ea1254a5d177
[    0.490711] clocksource: Switched to clocksource hyperv_clocksource_tsc_page
[    0.601944] VFS: Disk quotas dquot_6.6.0
[    0.603522] VFS: Dquot-cache hash table entries: 512 (order 0, 4096 bytes)
[    0.606131] FS-Cache: Loaded
[    0.607338] pnp: PnP ACPI init
[    0.608659] pnp 00:00: Plug and Play ACPI device, IDs PNP0501 (active)
[    0.611215] pnp 00:01: Plug and Play ACPI device, IDs PNP0501 (active)
[    0.613684] pnp 00:02: Plug and Play ACPI device, IDs PNP0b00 (active)
[    0.616242] pnp: PnP ACPI: found 3 devices
[    0.622516] NET: Registered protocol family 2
[    0.624504] tcp_listen_portaddr_hash hash table entries: 16384 (order: 6, 262144 bytes, linear)
[    0.627915] TCP established hash table entries: 262144 (order: 9, 2097152 bytes, linear)
[    0.631331] TCP bind hash table entries: 65536 (order: 8, 1048576 bytes, linear)
[    0.634148] TCP: Hash tables configured (established 262144 bind 65536)
[    0.636860] UDP hash table entries: 16384 (order: 7, 524288 bytes, linear)
[    0.639488] UDP-Lite hash table entries: 16384 (order: 7, 524288 bytes, linear)
[    0.642307] NET: Registered protocol family 1
[    0.644488] RPC: Registered named UNIX socket transport module.
[    0.646803] RPC: Registered udp transport module.
[    0.648577] RPC: Registered tcp transport module.
[    0.650349] RPC: Registered tcp NFSv4.1 backchannel transport module.
[    0.652757] PCI: CLS 0 bytes, default 64
[    0.654341] Trying to unpack rootfs image as initramfs...
[    0.656555] Freeing initrd memory: 60K
[    0.658011] PCI-DMA: Using software bounce buffering for IO (SWIOTLB)
[    0.660475] software IO TLB: mapped [mem 0x00000000f4000000-0x00000000f8000000] (64MB)
[    0.663565] KVM: vmx: using Hyper-V Enlightened VMCS
[    0.668754] kvm: already loaded the other module
[    0.673311] Initialise system trusted keyrings
[    0.675235] workingset: timestamp_bits=46 max_order=23 bucket_order=0
[    0.678598] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.681048] NFS: Registering the id_resolver key type
[    0.683002] Key type id_resolver registered
[    0.684603] Key type id_legacy registered
[    0.686199] Installing knfsd (copyright (C) 1996 okir@monad.swb.de).
[    0.689359] Key type cifs.idmap registered
[    0.691195] fuse: init (API version 7.32)
[    0.693029] SGI XFS with ACLs, security attributes, realtime, scrub, repair, quota, no debug enabled
[    0.697041] 9p: Installing v9fs 9p2000 file system support
[    0.699181] FS-Cache: Netfs '9p' registered for caching
[    0.701204] FS-Cache: Netfs 'ceph' registered for caching
[    0.703286] ceph: loaded (mds proto 32)
[    0.711691] NET: Registered protocol family 38
[    0.713503] xor: automatically using best checksumming function   avx
[    0.716233] Key type asymmetric registered
[    0.717870] Asymmetric key parser 'x509' registered
[    0.719763] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 250)
[    0.723684] hv_vmbus: registering driver hv_pci
[    0.725929] hv_pci fe0ab83c-a509-4bca-a91b-ea83adb286a7: PCI VMBus probing: Using version 0x10003
[    0.730295] hv_pci fe0ab83c-a509-4bca-a91b-ea83adb286a7: PCI host bridge to bus a509:00
[    0.733902] pci a509:00:00.0: [1414:008e] type 00 class 0x030200
[    0.741378] Serial: 8250/16550 driver, 4 ports, IRQ sharing disabled
[    0.744969] 00:00: ttyS0 at I/O 0x3f8 (irq = 4, base_baud = 115200) is a 16550A
[    0.748815] 00:01: ttyS1 at I/O 0x2f8 (irq = 3, base_baud = 115200) is a 16550A
[    0.752546] Non-volatile memory driver v1.3
[    0.759302] brd: module loaded
[    0.761493] loop: module loaded
[    0.763044] hv_vmbus: registering driver hv_storvsc
[    0.765356] wireguard: WireGuard 1.0.0 loaded. See www.wireguard.com for information.
[    0.766179] scsi host0: storvsc_host_t
[    0.768419] wireguard: Copyright (C) 2015-2019 Jason A. Donenfeld <Jason@zx2c4.com>. All Rights Reserved.
[    0.773464] tun: Universal TUN/TAP device driver, 1.6
[    0.775580] PPP generic driver version 2.4.2
[    0.777485] PPP BSD Compression module registered
[    0.779341] PPP Deflate Compression module registered
[    0.781337] PPP MPPE Compression module registered
[    0.783166] NET: Registered protocol family 24
[    0.784866] hv_vmbus: registering driver hv_netvsc
[    0.786932] VFIO - User Level meta-driver version: 0.3
[    0.789141] hv_vmbus: registering driver hyperv_keyboard
[    0.791304] rtc_cmos 00:02: RTC can wake from S4
[    0.794160] rtc_cmos 00:02: registered as rtc0
[    0.796156] rtc_cmos 00:02: setting system clock to 2021-06-26T13:40:34 UTC (1624714834)
[    0.799183] rtc_cmos 00:02: alarms up to one month, 114 bytes nvram
[    0.801814] device-mapper: ioctl: 4.43.0-ioctl (2020-10-01) initialised: dm-devel@redhat.com
[    0.805210] device-mapper: raid: Loading target version 1.15.1
[    0.807518] hv_utils: Registering HyperV Utility Driver
[    0.809556] hv_vmbus: registering driver hv_utils
[    0.811379] hv_vmbus: registering driver hv_balloon
[    0.811382] hv_utils: cannot register PTP clock: 0
[    0.813278] hv_vmbus: registering driver dxgkrnl
[    0.813891] hv_balloon: Using Dynamic Memory protocol version 2.0
[    0.815343] hv_utils: TimeSync IC version 4.0
[    0.817260] (NULL device *): dxgk: dxg_drv_init  Version: 2103
[    0.819549] Free page reporting enabled
[    0.823030] (NULL device *): dxgk: mmio allocated 9ffe00000  200000000 9ffe00000 bffdfffff
[    0.823862] hv_balloon: Cold memory discard hint enabled
[    0.827208] drop_monitor: Initializing network drop monitor service
[    0.833134] Mirror/redirect action on
[    0.834754] IPVS: Registered protocols (TCP, UDP)
[    0.836645] IPVS: Connection hash table configured (size=4096, memory=64Kbytes)
[    0.839510] IPVS: ipvs loaded.
[    0.840707] IPVS: [rr] scheduler registered.
[    0.842356] IPVS: [wrr] scheduler registered.
[    0.844012] IPVS: [sh] scheduler registered.
[    0.845711] ipip: IPv4 and MPLS over IPv4 tunneling driver
[    0.849528] ipt_CLUSTERIP: ClusterIP Version 0.8 loaded successfully
[    0.852491] Initializing XFRM netlink socket
[    0.854189] NET: Registered protocol family 10
[    0.856432] Segment Routing with IPv6
[    0.859317] sit: IPv6, IPv4 and MPLS over IPv4 tunneling driver
[    0.861689] NET: Registered protocol family 17
[    0.863404] Bridge firewalling registered
[    0.865007] 8021q: 802.1Q VLAN Support v1.8
[    0.866665] sctp: Hash tables configured (bind 512/512)
[    0.868791] 9pnet: Installing 9P2000 support
[    0.870494] Key type dns_resolver registered
[    0.872155] Key type ceph registered
[    0.873682] libceph: loaded (mon/osd proto 15/24)
[    0.875704] NET: Registered protocol family 40
[    0.877456] hv_vmbus: registering driver hv_sock
[    0.879286] IPI shorthand broadcast: enabled
[    0.881114] registered taskstats version 1
[    0.882719] Loading compiled-in X.509 certificates
[    0.884800] Btrfs loaded, crc32c=crc32c-generic
[    0.886942] Unstable clock detected, switching default tracing clock to "global"
[    0.886942] If you want to keep using the local clock, then add:
[    0.886942]   "trace_clock=local"
[    0.886942] on the kernel command line
[    0.895989] Freeing unused kernel image (initmem) memory: 1444K
[    0.934948] Write protecting the kernel read-only data: 22528k
[    0.938008] Freeing unused kernel image (text/rodata gap) memory: 2028K
[    0.940843] Freeing unused kernel image (rodata/data gap) memory: 632K
[    0.943306] Run /init as init process
[    0.944683]   with arguments:
[    0.945796]     /init
[    0.946676]   with environment:
[    0.947868]     HOME=/
[    0.948779]     TERM=linux
[    1.596511] scsi 0:0:0:0: Direct-Access     Msft     Virtual Disk     1.0  PQ: 0 ANSI: 5
[    1.600472] sd 0:0:0:0: Attached scsi generic sg0 type 0
[    1.601093] sd 0:0:0:0: [sda] 536870912 512-byte logical blocks: (275 GB/256 GiB)
[    1.603469] scsi 0:0:0:1: Direct-Access     Msft     Virtual Disk     1.0  PQ: 0 ANSI: 5
[    1.606555] sd 0:0:0:0: [sda] 4096-byte physical blocks
[    1.610279] sd 0:0:0:1: Attached scsi generic sg1 type 0
[    1.612276] sd 0:0:0:0: [sda] Write Protect is off
[    1.614900] sd 0:0:0:1: [sdb] 715632 512-byte logical blocks: (366 MB/349 MiB)
[    1.616558] sd 0:0:0:0: [sda] Mode Sense: 0f 00 00 00
[    1.619568] sd 0:0:0:1: [sdb] Write Protect is on
[    1.622051] sd 0:0:0:0: [sda] Write cache: enabled, read cache: enabled, doesn't support DPO or FUA
[    1.624058] sd 0:0:0:1: [sdb] Mode Sense: 0f 00 80 00
[    1.628790] sd 0:0:0:0: [sda] Attached SCSI disk
[    1.629729] sd 0:0:0:1: [sdb] Write cache: disabled, read cache: enabled, doesn't support DPO or FUA
[    1.636637] sd 0:0:0:1: [sdb] Attached SCSI disk
[    1.652312] hv_pci 15930f87-eaed-4b79-8753-551505091433: PCI VMBus probing: Using version 0x10003
[    1.657126] hv_pci 15930f87-eaed-4b79-8753-551505091433: PCI host bridge to bus eaed:00
[    1.660009] pci_bus eaed:00: root bus resource [mem 0xc00000000-0xe00001fff window]
[    1.663662] pci eaed:00:00.0: [1af4:105a] type 00 class 0x088000
[    1.667859] pci eaed:00:00.0: reg 0x10: [mem 0xe00000000-0xe00000fff 64bit]
[    1.672074] pci eaed:00:00.0: reg 0x18: [mem 0xe00001000-0xe00001fff 64bit]
[    1.676384] pci eaed:00:00.0: reg 0x20: [mem 0xc00000000-0xdffffffff 64bit]
[    1.683042] pci eaed:00:00.0: BAR 4: assigned [mem 0xc00000000-0xdffffffff 64bit]
[    1.687260] pci eaed:00:00.0: BAR 0: assigned [mem 0xe00000000-0xe00000fff 64bit]
[    1.691570] pci eaed:00:00.0: BAR 2: assigned [mem 0xe00001000-0xe00001fff 64bit]
[    1.705456] virtiofs virtio0: Cache len: 0x200000000 @ 0xc00000000
[    1.769727] memmap_init_zone_device initialised 2097152 pages in 20ms
[    1.777472] EXT4-fs (sdb): mounted filesystem without journal. Opts: (null)
[    1.784918] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[    1.786400] scsi 0:0:0:2: Direct-Access     Msft     Virtual Disk     1.0  PQ: 0 ANSI: 5
[    1.787385] EXT4-fs (sda): mounted filesystem with ordered data mode. Opts: (null)
[    1.793790] sd 0:0:0:2: Attached scsi generic sg2 type 0
[    1.794125] sd 0:0:0:2: [sdc] 536870912 512-byte logical blocks: (275 GB/256 GiB)
[    1.798821] sd 0:0:0:2: [sdc] 4096-byte physical blocks
[    1.800868] sd 0:0:0:2: [sdc] Write Protect is off
[    1.803081] sd 0:0:0:2: [sdc] Mode Sense: 0f 00 00 00
[    1.805740] sd 0:0:0:2: [sdc] Write cache: enabled, read cache: enabled, doesn't support DPO or FUA
[    1.810470] sd 0:0:0:2: [sdc] Attached SCSI disk
[    1.874928] Adding 8388608k swap on /swap/file.  Priority:-2 extents:4 across:8413184k
[    1.882816] EXT4-fs (sdc): recovery complete
[    1.885224] EXT4-fs (sdc): mounted filesystem with ordered data mode. Opts: discard,errors=remount-ro,data=ordered
[    1.917247] hv_pci ae3bdc20-b9ca-4291-90d5-c2b0cbbc4e51: PCI VMBus probing: Using version 0x10003
[    1.972059] hv_pci ae3bdc20-b9ca-4291-90d5-c2b0cbbc4e51: PCI host bridge to bus b9ca:00
[    1.975368] pci_bus b9ca:00: root bus resource [mem 0xbffe00000-0xbffe02fff window]
[    1.979056] pci b9ca:00:00.0: [1af4:1049] type 00 class 0x010000
[    1.982090] pci b9ca:00:00.0: reg 0x10: [mem 0xbffe00000-0xbffe00fff 64bit]
[    1.985244] pci b9ca:00:00.0: reg 0x18: [mem 0xbffe01000-0xbffe01fff 64bit]
[    1.988347] pci b9ca:00:00.0: reg 0x20: [mem 0xbffe02000-0xbffe02fff 64bit]
[    1.993922] pci b9ca:00:00.0: BAR 0: assigned [mem 0xbffe00000-0xbffe00fff 64bit]
[    1.997168] pci b9ca:00:00.0: BAR 2: assigned [mem 0xbffe01000-0xbffe01fff 64bit]
[    2.000319] pci b9ca:00:00.0: BAR 4: assigned [mem 0xbffe02000-0xbffe02fff 64bit]
[    2.294358] hv_pci ff76c621-4dbb-4cca-a959-f47f82d0627f: PCI VMBus probing: Using version 0x10003
[    2.295487] 9pnet_virtio: no channels available for device drvfs
[    2.300658] WARNING: mount: waiting for virtio device...
[    2.349412] hv_pci ff76c621-4dbb-4cca-a959-f47f82d0627f: PCI host bridge to bus 4dbb:00
[    2.352743] pci_bus 4dbb:00: root bus resource [mem 0xbffe04000-0xbffe06fff window]
[    2.356436] pci 4dbb:00:00.0: [1af4:1049] type 00 class 0x010000
[    2.359481] pci 4dbb:00:00.0: reg 0x10: [mem 0xbffe04000-0xbffe04fff 64bit]
[    2.362579] pci 4dbb:00:00.0: reg 0x18: [mem 0xbffe05000-0xbffe05fff 64bit]
[    2.365699] pci 4dbb:00:00.0: reg 0x20: [mem 0xbffe06000-0xbffe06fff 64bit]
[    2.371224] pci 4dbb:00:00.0: BAR 0: assigned [mem 0xbffe04000-0xbffe04fff 64bit]
[    2.374432] pci 4dbb:00:00.0: BAR 2: assigned [mem 0xbffe05000-0xbffe05fff 64bit]
[    2.377590] pci 4dbb:00:00.0: BAR 4: assigned [mem 0xbffe06000-0xbffe06fff 64bit]
[    2.407505] hv_pci e9efebb1-e223-44f3-8962-926dfae8df95: PCI VMBus probing: Using version 0x10003
[    2.462078] hv_pci e9efebb1-e223-44f3-8962-926dfae8df95: PCI host bridge to bus e223:00
[    2.465260] pci_bus e223:00: root bus resource [mem 0xbffe08000-0xbffe0afff window]
[    2.468879] pci e223:00:00.0: [1af4:1049] type 00 class 0x010000
[    2.471865] pci e223:00:00.0: reg 0x10: [mem 0xbffe08000-0xbffe08fff 64bit]
[    2.474994] pci e223:00:00.0: reg 0x18: [mem 0xbffe09000-0xbffe09fff 64bit]
[    2.478138] pci e223:00:00.0: reg 0x20: [mem 0xbffe0a000-0xbffe0afff 64bit]
[    2.483669] pci e223:00:00.0: BAR 0: assigned [mem 0xbffe08000-0xbffe08fff 64bit]
[    2.486792] pci e223:00:00.0: BAR 2: assigned [mem 0xbffe09000-0xbffe09fff 64bit]
[    2.489955] pci e223:00:00.0: BAR 4: assigned [mem 0xbffe0a000-0xbffe0afff 64bit]
[    2.509202] FS-Cache: Duplicate cookie detected
[    2.511064] FS-Cache: O-cookie c=00000000e665a4ed [p=00000000ca497d02 fl=222 nc=0 na=1]
[    2.514264] FS-Cache: O-cookie d=00000000e34441ad n=00000000d8f76ff4
[    2.516901] FS-Cache: O-key=[10] '34323934393337353133'
[    2.518946] FS-Cache: N-cookie c=000000003f965cd8 [p=00000000ca497d02 fl=2 nc=0 na=1]
[    2.521927] FS-Cache: N-cookie d=00000000e34441ad n=00000000a81afc88
[    2.524366] FS-Cache: N-key=[10] '34323934393337353133'
[    2.685755] hv_pci ea0c0e95-e1e8-4c48-919c-57898404e91f: PCI VMBus probing: Using version 0x10003
[    2.687242] 9pnet_virtio: no channels available for device drvfs
[    2.691622] WARNING: mount: waiting for virtio device...
[    2.740339] hv_pci ea0c0e95-e1e8-4c48-919c-57898404e91f: PCI host bridge to bus e1e8:00
[    2.743334] pci_bus e1e8:00: root bus resource [mem 0xbffe0c000-0xbffe0efff window]
[    2.747100] pci e1e8:00:00.0: [1af4:1049] type 00 class 0x010000
[    2.750227] pci e1e8:00:00.0: reg 0x10: [mem 0xbffe0c000-0xbffe0cfff 64bit]
[    2.753393] pci e1e8:00:00.0: reg 0x18: [mem 0xbffe0d000-0xbffe0dfff 64bit]
[    2.756636] pci e1e8:00:00.0: reg 0x20: [mem 0xbffe0e000-0xbffe0efff 64bit]
[    2.762360] pci e1e8:00:00.0: BAR 0: assigned [mem 0xbffe0c000-0xbffe0cfff 64bit]
[    2.765680] pci e1e8:00:00.0: BAR 2: assigned [mem 0xbffe0d000-0xbffe0dfff 64bit]
[    2.768929] pci e1e8:00:00.0: BAR 4: assigned [mem 0xbffe0e000-0xbffe0efff 64bit]
[    2.798096] hv_pci a603022f-41ff-47ad-9991-067be9796c5f: PCI VMBus probing: Using version 0x10003
[    2.852537] hv_pci a603022f-41ff-47ad-9991-067be9796c5f: PCI host bridge to bus 41ff:00
[    2.855632] pci_bus 41ff:00: root bus resource [mem 0xbffe10000-0xbffe12fff window]
[    2.859312] pci 41ff:00:00.0: [1af4:1049] type 00 class 0x010000
[    2.862321] pci 41ff:00:00.0: reg 0x10: [mem 0xbffe10000-0xbffe10fff 64bit]
[    2.865589] pci 41ff:00:00.0: reg 0x18: [mem 0xbffe11000-0xbffe11fff 64bit]
[    2.868754] pci 41ff:00:00.0: reg 0x20: [mem 0xbffe12000-0xbffe12fff 64bit]
[    2.874311] pci 41ff:00:00.0: BAR 0: assigned [mem 0xbffe10000-0xbffe10fff 64bit]
[    2.877570] pci 41ff:00:00.0: BAR 2: assigned [mem 0xbffe11000-0xbffe11fff 64bit]
[    2.880789] pci 41ff:00:00.0: BAR 4: assigned [mem 0xbffe12000-0xbffe12fff 64bit]
[   49.665618] hv_balloon: Max. dynamic memory size: 32702 MB
[   77.827532] kvm: exiting hardware virtualization
[   77.830276] sd 0:0:0:2: [sdc] Synchronizing SCSI cache
[   77.832757] sd 0:0:0:0: [sda] Synchronizing SCSI cache
[   77.835479] ACPI: Preparing to enter system sleep state S5
[   77.837739] reboot: Power down
[   77.844147] acpi_power_off called
WSL has terminated, press any key to continue...
```

**Comment by jodavaho** [comment]
What "Tasks" remain (above on toolbar?)

**Comment by parsec** [comment]
> 
> 
> What "Tasks" remain (above on toolbar?)

It's because the template has a checkbox for the WSL version. It thinks it's a task list.

**Comment by Steffanic** [comment]
Have you tried a full PC restart? Try to restart from the start menu, but hold shift when you press, "Restart." This fixed my problem. I figured it out by trying to run Ubuntu from the wsl --system command. It spat out this error: "The operation timed out because a response was not received from the virtual machine or container.", which led me [here](https://superuser.com/questions/1634793/wsl-ubuntu-bash-the-operation-timed-out-because-a-response-was-not-received-fro). Hopefully it helps!

**Comment by jodavaho** [comment]
A restart temporarily solves the problem -- In that I can then access WSL -- but doesn't really "solve" the problem. Hangs return.

**Comment by Steffanic** [comment]
I have moved my WSL distro(the vhdx file) to a new drive by exporting it, then importing it(like in [this article](https://stackoverflow.com/questions/38779801/move-wsl-bash-on-windows-root-filesystem-to-another-hard-drive).) And I have not experienced any more hangs. This is nothing short of a hack and I think that there is probably something weird happening in WSL. Hopefully this "fixes" it.

**Comment by tsmvision** [comment]
In my case, whenever the WSL2 terminal hangs when returning from the sleep mode

* go to services
* find LxssManager
* right click and choose restart

It fixes the issues but looking for a permanent solution than this kind of patchwork.

**Comment by Sillians** [comment]
> 

I do make it work using the above steps, but recently it hangs and doesn't re-start.

**Comment by pdowson-rsc** [comment]
My team is seeing this issue, personally I have seen it on Windows 11. My steps to resolve are:

1. Check that LxssManager is in the status "Stopping" in `Services.msc`
3. Make sure I have any development tools closed
4. Open `Powershell` as administrator
5. Determine the PID of the LxssManager using `tasklist /svc /fi "imagename eq svchost.exe" | Select-String 'LxssManager'`
6. Using `procexp.exe` running as an administrator, locate the PID and stop the process. I find it easier to lookup the PID using powershell.
7. Reopen WSL and/or Docker Desktop and LxssManager starts normally

NOTE 1: Attempting to stop the process in Powershell using `Get-Process -Id 3972 | Stop-Process -Force` returns an access denied even as an elevated administrator

NOTE 2: Attempting to stop the process in Services.msc has no effect

NOTE 3: I abandoned using the built-in task manager, not sure if that will also kill the process or not.

**Comment by Hyper200** [comment]
I am also having this issue as of recently - My question is do any of you have anything like beyond privilege management (Or a PAM Auth soloution of any kind?) My issues seem to have started after that.

**Comment by OneBlue** [comment]
This might be a duplicate of #8170. Can someone collect log and share repro instruction if this is still happening with the latest Windows updates ?

**Comment by Hyper200** [comment]
The PAM Privlage magament stuff i has severly broke windows - It ended up getting re-installed by our local IT, and i never found a soloution sorry @remoford

**Comment by OneBlue** [comment]
Thank you @remoford. 

Unfortunately I'm not seeing anything that jumps out. 

Next time this happens, can you please share a dump of the LxssManager process ?

**Comment by leandropereira-hotmart** [comment]
Hi guys. I RESOLVED!!!

After a lot of research, I found a suggest of try kill process by Process Explorer
(you can download here https://learn.microsoft.com/en-us/sysinternals/downloads/process-explorer)...

**Open Process Explorer as Admin and find the service LxssManager between the services called "svchost.exe"**

**(I could'nt found by PID, so I checked svchost.exe one by one until find)**

**Comment by sql-sith** [comment]
I also have the same problem. When it occurs, I tell the lxssmanager service to stop (which hangs), then find the PID and kill its process tree in task manager.

I don't know if this will help, but the last time I was in this state, before stopping the service and killing the PID, I ran the `waitingon.exe` utility on each `wsl` process on my laptop. They all reported that they were waiting on PID 14692, which was lxssmanager, and they all had the same message: 

```
Thread ID: 39040 is waiting on a Alpc (ObjectName: \RPC Control\OLED0D0EECC34DA561C57A33C21644E) which is currently owned by Process ID: 14692 ()
```
Does this message give anyone insight into what might be going on?

The full output of `waitingon.exe` follows. My apologies for not running from an elevated PowerShell.

```
PS > (get-process wsl) | ForEach-Object { C:\util\WaitingOn.exe $_.Id }

Written by Leonardo Fagundes. No rights Reserved.

Warning: WaitingOn.exe is not elevated. Only processes for the current user will be analyzed.

Enumerating all blocked threads for the specified process...


Process ID: 11568 (wsl.exe)
    └-Thread ID: 11016 is waiting on a Alpc (ObjectName: \RPC Control\OLED0D0EECC34DA561C57A33C21644E) which is currently owned by Process ID: 14692 ()

Written by Leonardo Fagundes. No rights Reserved.

Warning: WaitingOn.exe is not elevated. Only processes for the current user will be analyzed.

Enumerating all blocked threads for the specified process...


Process ID: 15260 (wsl.exe)
    └-Thread ID: 39040 is waiting on a Alpc (ObjectName: \RPC Control\OLED0D0EECC34DA561C57A33C21644E) which is currently owned by Process ID: 14692 ()

Written by Leonardo Fagundes. No rights Reserved.

Warning: WaitingOn.exe is not elevated. Only processes for the current user will be analyzed.

Enumerating all blocked threads for the specified process...


Process ID: 17276 (wsl.exe)
    └-Thread ID: 38484 is waiting on a Alpc (ObjectName: \RPC Control\OLED0D0EECC34DA561C57A33C21644E) which is currently owned by Process ID: 14692 ()

Written by Leonardo Fagundes. No rights Reserved.

Warning: WaitingOn.exe is not elevated. Only processes for the current user will be analyzed.

Enumerating all blocked threads for the specified process...


Process ID: 17804 (wsl.exe)
    └-Thread ID: 40748 is waiting on a Alpc (ObjectName: \RPC Control\OLED0D0EECC34DA561C57A33C21644E) which is currently owned by Process ID: 14692 ()

Written by Leonardo Fagundes. No rights Reserved.

Warning: WaitingOn.exe is not elevated. Only processes for the current user will be analyzed.

Enumerating all blocked threads for the specified process...


Process ID: 18668 (wsl.exe)
    └-Thread ID: 38328 is waiting on a Alpc (ObjectName: \RPC Control\OLED0D0EECC34DA561C57A33C21644E) which is currently owned by Process ID: 14692 ()

Written by Leonardo Fagundes. No rights Reserved.

Warning: WaitingOn.exe is not elevated. Only processes for the current user will be analyzed.

Enumerating all blocked threads for the specified process...


Process ID: 30632 (wsl.exe)
    └-Thread ID: 38188 is waiting on a Alpc (ObjectName: \RPC Control\OLED0D0EECC34DA561C57A33C21644E) which is currently owned by Process ID: 14692 ()

Written by Leonardo Fagundes. No rights Reserved.

Warning: WaitingOn.exe is not elevated. Only processes for the current user will be analyzed.

Enumerating all blocked threads for the specified process...


Process ID: 30828 (wsl.exe)
    └-Thread ID: 27516 is waiting on a Alpc (ObjectName: \RPC Control\OLED0D0EECC34DA561C57A33C21644E) which is currently owned by Process ID: 14692 ()

Written by Leonardo Fagundes. No rights Reserved.

Warning: WaitingOn.exe is not elevated. Only processes for the current user will be analyzed.

Enumerating all blocked threads for the specified process...


Process ID: 31208 (wsl.exe)
    └-Thread ID: 34444 is waiting on a Alpc (ObjectName: \RPC Control\OLED0D0EECC34DA561C57A33C21644E) which is currently owned by Process ID: 14692 ()

Written by Leonardo Fagundes. No rights Reserved.

Warning: WaitingOn.exe is not elevated. Only processes for the current user will be analyzed.

Enumerating all blocked threads for the specified process...


Process ID: 35472 (wsl.exe)
    └-Thread ID: 30916 is waiting on a Alpc (ObjectName: \RPC Control\OLED0D0EECC34DA561C57A33C21644E) which is currently owned by Process ID: 14692 ()

Written by Leonardo Fagundes. No rights Reserved.

Warning: WaitingOn.exe is not elevated. Only processes for the current user will be analyzed.

Enumerating all blocked threads for the specified process...


Process ID: 35912 (wsl.exe)
    └-Thread ID: 23804 is waiting on a Alpc (ObjectName: \RPC Control\OLED0D0EECC34DA561C57A33C21644E) which is currently owned by Process ID: 14692 ()

Written by Leonardo Fagundes. No rights Reserved.

Warning: WaitingOn.exe is not elevated. Only processes for the current user will be analyzed.

Enumerating all blocked threads for the specified process...


Process ID: 37088 (wsl.exe)
    └-Thread ID: 36436 is waiting on a Alpc (ObjectName: \RPC Control\OLED0D0EECC34DA561C57A33C21644E) which is currently owned by Process ID: 14692 ()

Written by Leonardo Fagundes. No rights Reserved.

Warning: WaitingOn.exe is not elevated. Only processes for the current user will be analyzed.

Enumerating all blocked threads for the specified process...


Process ID: 39892 (wsl.exe)
    └-Thread ID: 39012 is waiting on a Alpc (ObjectName: \RPC Control\OLED0D0EECC34DA561C57A33C21644E) which is currently owned by Process ID: 14692 ()
```
